src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/JsonSourceGeneratorDiagnosticsTests.cs

```diff
@@ -126,60 +126,58 @@ public class IndexViewModel
 
             JsonSourceGeneratorResult result = CompilationHelper.RunJsonSourceGenerator(compilation);
 
+            Assert.Empty(result.NewCompilation.GetDiagnostics());
             Assert.Empty(result.Diagnostics);
         }
 
-        [Fact]
-        public void MultiDimensionArrayDoesNotProduceWarnings()
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void MultiDimensionArrayDoesNotProduceWarnings(bool explicitRef)
         {
-            static void RunTest(bool explicitRef)
-            {
-                // Compile the referenced assembly first.
-                Compilation campaignCompilation = CompilationHelper.CreateCampaignSummaryViewModelCompilation();
-                Compilation eventCompilation = CompilationHelper.CreateActiveOrUpcomingEventCompilation();
+            // Compile the referenced assembly first.
```

Comment:
These diffs fix a couple of tests that were actually failing to compile but we weren't checking for that. Discovered by the newly added diagnostics.